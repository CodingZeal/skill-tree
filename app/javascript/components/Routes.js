import React from "react"
import ReactDOM from 'react-dom'
import PropTypes from "prop-types"
import {BrowserRouter as Router, Route} from "react-router-dom";
import StaticProfile from './StaticProfile'
import Profile from './Profile'

class Routes extends React.Component {
  constructor(props){
    super(props)
      this.state = {
        profile: {
          firstname: 'John',
          lastname: 'Doe',
          email: 'john@example.com',
          timezone: 'UTC',
          url: 'iou22o38f4',
        },
      }
    }

  render () {
      const {profile} = this.state

      // builds custom url
      const profUrl = `/profile/${profile.url}`

    return (
      <Router>
        <Route path="/profile" exact render={(props) => <Profile{...props}
          profile={profile}
        />} />
        <Route path={profUrl} exact render={(props) => <StaticProfile{...props}
          profile={profile}
        />} />
      </Router>
    );
  }
}

export default Routes
