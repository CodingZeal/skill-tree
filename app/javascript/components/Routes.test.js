import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {BrowserRouter as Router, Route} from "react-router-dom";
import Routes from './Routes';

Enzyme.configure({ adapter: new Adapter() });

describe('<Routes>', () => {
  const wrapper = shallow(<Routes />);
    it('should render corretly', () => {
        expect(wrapper).toMatchSnapshot()
    })
    it('should have a router', () => {
      expect(wrapper.find(Router)).toHaveLength(1)
    })
    it('has a route to the dashboard (<Profile/>)', () => {
      expect(wrapper.find('[path="/profile"]')).toHaveLength(1)
    })
})
