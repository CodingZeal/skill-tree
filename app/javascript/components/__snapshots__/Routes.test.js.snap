// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes> should render corretly 1`] = `ShallowWrapper {}`;
